tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py

```diff
@@ -182,109 +187,114 @@ def test__get_only_private_collections_with_auth__OK(self):
+                ],
+                "tissue": [{"label": "test_tissue", "ontology_term_id": "test_obo"}],
+                "tombstone": False,
+                "x_approximate_distribution": "NORMAL",
+                "x_normalization": "test_x_normalization",
+            }
+        ],
+        "description": "test_description",
+        "id": "test_collection_id",
+        "links": [
+            {"link_name": "test_doi_link_name", "link_type": "DOI", "link_url": "http://test_doi_url.place"},
+            {"link_name": None, "link_type": "DOI", "link_url": "http://test_no_link_name_doi_url.place"},
+            {
+                "link_name": "test_raw_data_link_name",
+                "link_type": "RAW_DATA",
+                "link_url": "http://test_raw_data_url.place",
+            },
+            {"link_name": None, "link_type": "RAW_DATA", "link_url": "http://test_no_link_name_raw_data_url.place"},
+            {
+                "link_name": "test_protocol_link_name",
+                "link_type": "PROTOCOL",
+                "link_url": "http://test_protocol_url.place",
+            },
+            {"link_name": None, "link_type": "PROTOCOL", "link_url": "http://test_no_link_name_protocol_url.place"},
+            {
+                "link_name": "test_lab_website_link_name",
+                "link_type": "LAB_WEBSITE",
+                "link_url": "http://test_lab_website_url.place",
+            },
+            {
+                "link_name": None,
+                "link_type": "LAB_WEBSITE",
+                "link_url": "http://test_no_link_name_lab_website_url.place",
+            },
+            {"link_name": "test_other_link_name", "link_type": "OTHER", "link_url": "http://test_other_url.place"},
+            {"link_name": None, "link_type": "OTHER", "link_url": "http://test_no_link_name_other_url.place"},
+            {
+                "link_name": "test_data_source_link_name",
+                "link_type": "DATA_SOURCE",
+                "link_url": "http://test_data_source_url.place",
+            },
+            {
+                "link_name": None,
+                "link_type": "DATA_SOURCE",
+                "link_url": "http://test_no_link_name_data_source_url.place",
+            },
+        ],
+        "name": "test_collection_name",
+        "published_at": None,
+        "publisher_metadata": None,
+        "revised_at": None,
+        "revision_of": None,
+        "tombstone": False,
+        "visibility": "PUBLIC",
+    }
+
     def setUp(self):
         super().setUp()
         self.test_collection = dict(
             name="collection", description="description", contact_name="john doe", contact_email="johndoe@email.com"
         )
 
-    def test__get_public_collection_verify_body__OK(self):
+    def test__get_public_collection_verify_body_is_reshaped_correctly__OK(self):
+        dataset = self.session.query(DbDataset).filter(DbDataset.id == "test_dataset_id").one_or_none()
```

Comment:
Why do you need to change it to a dict and is this related to the PR or extra?

Reply:
I'll add more comments to explain this.

It's to make sure that the reshaping function correctly converts lone dicts to _lists_ of dicts (of length 1).

The return type needs to be an array.